app/models/adapters/payment_shared_adapter.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+module Adapters
+  class PaymentSharedAdapter
+    def process_payment(payment)
+      {
+        pay_check_dt: payment[:payment_date],
+        pay_check_amount: ActiveSupport::NumberHelper.number_to_currency(payment[:payment_amount]),
+        pay_check_type: payment[:payment_type],
+        payment_method: get_payment_method(payment),
+        bank_name: payment.dig(:address_eft, :bank_name),
+        account_number: mask_account_number(payment[:address_eft])
+      }
+    end
+
+    def mask_account_number(address_eft, all_but = 4, char = '*')
+      return if address_eft.blank?
+
+      account_number = address_eft[:account_number]
+      return if account_number.blank?
+
+      account_number&.gsub(/.(?=.{#{all_but}})/, char)
```

Comment:
This safety operator shouldn't be necessary due to the guard above.

Reply:
This part is from the va.gov api which is why I didn't tamper with it outside of moving it to a shared adapter to use it within the mobile api (to reduce duplicate code).